lrlex/src/lib/lexer.rs

```diff
@@ -259,6 +259,30 @@ impl<'a, StorageT: Copy + Eq + Hash + PrimInt + Unsigned> Lexer<StorageT>
         )
     }
 
+    fn surrounding_line_str(&self, i: usize) -> &str {
+        fn surrounding_line_off<StorageT>(lexer: &LRLexer<StorageT>, i: usize) -> (usize, usize) {
```

Comment:
Why is this a function? Can't be just inline the code, since all the outer `surrounding_line_str` does is call the inner function?

Reply:
If you look at the logic inside, it's quite tricky to make it part of the outer function. [I actually did a version which inlined everything, but I found it much harder to follow.]

Reply:
Ah, I see what you mean now.